crates/l2/proposer/l1_committer.rs

```diff
@@ -276,16 +276,20 @@ impl Committer {
                 None => 0,
             };
 
+            let new_nonce = if account_update.info.is_none() {
+                prev_nonce
+            } else {
+                account_update
+                    .info
+                    .as_ref()
+                    .ok_or(CommitterError::FailedToRetrieveDataFromStorage)?
+                    .nonce
+            };
```

Comment:
Nit: it shouldn't be necessary to throw an error as you can match the `Option<AccountInfo>`:

```suggestion
            let nonce_diff = if let Some(info) = account_update.info {
                info.nonce - prev_nonce
            } else {
                0
            };
```

Reply:
or
```rust
let new_nonce = account_update.info.map(|info| info.nonce - prev_nonce).unwrap_or(0);
```
if you like oneliners

Reply:
I don't think defaulting to 0 is the right approach here.

Reply:
although if you address Fran's comment, the first approach should be easier for throwing an overflow error (it's possible doing it the functional way but it gets messy)

Reply:
> although if you address Fran's comment, the first approach should be easier for throwing an overflow error (it's possible doing it the functional way but it gets messy)

I personally found the oneliner a bit harder to read. I think I liked the first approach a bit better.

Reply:
@xqft Hey Tefo, what about this? https://github.com/lambdaclass/ethrex/pull/2133/commits/1f10efc2e96ac8591b7ac1f5e7146d90ee6fd362
It's a bit of an in-between. I liked the if let version better; but I maintained the substraction inside the construction of the Struct.

Reply:
@fkrause98 sorry just saw your response, why do you think that? I thought a `None` account info in an account update meant no change in account info by definition, therefore no change in nonce, therefore `nonce_diff = 0`

Reply:
@lima-limon-inc looks good!!

Reply:
> @fkrause98 sorry just saw your response, why do you think that? I thought a None account info in an account update meant no change in account info by definition, therefore no change in nonce, therefore nonce_diff = 0

I believe that the problem with defaulting to 0 is that it hides the cases where, for some reason, the nonce is negative. I think raising an error, were that to happen, is valuable.